lib/bolt/transport/sudoable/connection.rb

```diff
@@ -38,7 +39,8 @@ def make_executable(path)
 
         def make_tempdir
           tmpdir = @target.options.fetch('tmpdir', '/tmp')
-          tmppath = "#{tmpdir}/#{SecureRandom.uuid}"
+          script_dir = @target.options.fetch('script-dir', SecureRandom.uuid)
```

Comment:
I think we should also add some validation to this method: https://github.com/puppetlabs/bolt/blob/master/lib/bolt/transport/ssh.rb#L29 to validate that the option isn't empty. It should error there if it's an empty string.